server/sonar-search/src/main/java/org/sonar/search/SearchServer.java

```diff
@@ -50,56 +57,182 @@ public SearchServer(Props props) {
 
   @Override
   public void start() {
-    Jmx.register(EsSettingsMBean.OBJECT_NAME, settings);
-    initBootstrap();
-    Settings esSettings = settings.build();
-    if (esSettings.getAsInt(MIMINUM_MASTER_NODES, 1) >= 2) {
-      LOGGER.info("Elasticsearch is waiting {} for {} node(s) to be up to start.",
-        esSettings.get(INITIAL_STATE_TIMEOUT),
-        esSettings.get(MIMINUM_MASTER_NODES)
-      );
+    Path absolutePath = findExecutable();
+
+    List<String> command = new ArrayList<>();
+    command.add(absolutePath.toString());
+    Map<String, String> settingsMap = settings.build();
+    settingsMap.entrySet().stream()
+      .filter(entry -> !"path.home".equals(entry.getKey()))
+      .forEach(entry -> command.add("-E" + entry.getKey() + "=" + entry.getValue()));
+    url = "http://"+settingsMap.get("http.host") + ":" + settingsMap.get("http.port");
+    System.out.println(command.stream().collect(Collectors.joining(" ")));
+    ProcessBuilder builder = new ProcessBuilder(command)
+      .directory(new File(absolutePath.getParent().toString()));
+    builder.redirectOutput(ProcessBuilder.Redirect.PIPE);
+    builder.redirectErrorStream(true);
+    try {
+      p = builder.start();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    CountDownLatch latch = new CountDownLatch(2);
+
+    new Thread(() -> {
+      InputStream inputStream = p.getInputStream();
+      InputStreamReader reader1 = new InputStreamReader(inputStream);
+      BufferedReader reader = new BufferedReader(reader1);
+      String line;
+      try {
+        while ((line = reader.readLine()) != null) {
+          System.out.println(line);
+          if (line.contains(" publish_address ")) {
+            latch.countDown();
+          }
+          if (line.contains(" started")) {
+            latch.countDown();
+          }
+        }
+      } catch (IOException e) {
+        e.printStackTrace();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1148)